### PR TITLE
Add dockerfiles

### PR DESCRIPTION
**Pull request by rhdxmr**
- Add Dockerfile for Ubuntu 21.04
- Add Dockerfile for Apline 3.14
- Modify Dockerfile of fedora-35 for upgrading LLVM 12.0.1

### CODE REVIEW

**Comment by rhdxmr** [comment]
@rsdy Hello
Can you review this PR? Thanks,

File: .github/workflows/docker-publish.yml [hunk 1]
```diff
@@ -49,12 +59,32 @@ jobs:
...
-        id: rawhide
+      # - name: Build Fedora Rawhide
+      #   id: rawhide
+      #   run: |
+      #     VERSION=${{ env.version }}-fedora-rawhide
+      #     docker build --no-cache \
+      #                  -f Dockerfile.fedora \
+      #                  -t $IMAGE_NAME:$VERSION \
+      #                  .
+      #     echo "::set-output name=image_id::$IMAGE_NAME:$VERSION"
```
Comment (rsdy) [inline .github/workflows/docker-publish.yml]: I think it's ok to just delete this

File: Dockerfile.alpine [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    && VERS=1.0.2p \
+    && CHECKSUM=50a98e07b1a89eb8f6a99477f262df71c6fa7bef77df4dc83025a2845c827d00 \
+    && curl -sqO https://www.openssl.org/source/openssl-$VERS.tar.gz \
+    && echo "$CHECKSUM  openssl-$VERS.tar.gz" > checksums.txt \
+    && sha256sum -c checksums.txt \
+    && tar xzf openssl-$VERS.tar.gz && cd openssl-$VERS \
+    && ./Configure ${openssl_arch} -fPIC --prefix=/usr/local/musl/$TARGET \
+    && make depend \
+    && make && make install \
+    && cd .. && rm -rf openssl-$VERS.tar.gz openssl-$VERS checksums.txt
```
Comment (rsdy) [inline Dockerfile.alpine]: Since Alpine is musl anyway i think it's better to use the openssl version they ship in APK.

**Review by rsdy — APPROVED** [review]

**Comment by rhdxmr** [comment]
@rsdy I applied your corrections and re-pushed the branch. Thanks,

File: Dockerfile.alpine (hunk 2 above)
Comment (rhdxmr) [reply Dockerfile.alpine]: By the way, why is musl installed in ubuntu or fedora?

Comment (rsdy) [reply Dockerfile.alpine]: Since we generally want to be able to test building a static binary on those platforms, too

**Pull request merged**